src/test/kotlin/org/rust/ide/highlight/RsRainbowVisitorTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.highlight
+
+import org.rust.lang.RsTestBase
+
+class RsRainbowVisitorTest : RsTestBase() {
+
+    fun checkRainbow(code: String, isRainbowOn: Boolean = true, withColor: Boolean = false) {
+        myFixture.testRainbow(
+            "main.rs",
+            code,
+            isRainbowOn, withColor)
+    }
+
+    fun testPathBinding() = checkRainbow("""
+        fn main() {
+            let mut <rainbow>test</rainbow> = "";
+            <rainbow>test</rainbow> = "";
+        }
+    """)
+
+    fun testFunctionBinding() = checkRainbow("""
+        fn foo() {}
+        fn main() {
+            foo();
+        }
+    """, false)
+
+    fun testDifferentColor() = checkRainbow("""
+        fn main() {
+            let mut <rainbow color='ff000003'>test</rainbow> = "";
+            <rainbow color='ff000003'>test</rainbow> = "";
+            let mut <rainbow color='ff000004'>test</rainbow> = "";
+            <rainbow color='ff000004'>test</rainbow> = "";
+        }
+    """, withColor = true)
+
+    fun testComplexDifferentColor() = checkRainbow("""
+        fn foo(<rainbow color='ff000003'>param</rainbow>: i32) {
+            let <rainbow color='ff000003'>x</rainbow> = <rainbow color='ff000003'>param</rainbow> + <rainbow color='ff000003'>param</rainbow>;
+            let <rainbow color='ff000001'>y</rainbow> = {
+               let <rainbow color='ff000003'>param</rainbow> = <rainbow color='ff000003'>x</rainbow>;
```

Comment:
Hm, ` <rainbow color='ff000003'>param</rainbow> ` does not look right... It's not the same param as in the `fn foo(param: i32)`, so it should probably use a different color.